.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,68 @@
+name: Docker
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ main ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+    paths:
+      - 'examples/telegraf/**'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'examples/telegraf/**'
+
+  workflow_dispatch:
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Login against a Docker registry except on PR
```

Comment:
So, it would be super cool if we also pushed the images for pull requests. We do this for IPDK, and save them for a day. The advantage is that you can then pull those container images from GHCR and test them locally if you want.

Reply:
Didn't want to create too many docker images at the beginning, wanted to add them later on, once ci build is working and approved

Reply:
Keep in mind we have unlimited API access to GHCR, and also unlimited storage (but we should set the expiry on PR GHCR docker container pushes to 1 or 2 days).